File: src/main/java/com/vaadin/demo/component/login/LoginOverlayBasic.java
package com.vaadin.demo.component.login;

import com.vaadin.flow.component.button.Button;
import com.vaadin.flow.component.button.ButtonVariant;
import com.vaadin.flow.component.login.LoginOverlay;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.router.Route;
import com.vaadin.demo.DemoExporter; // hidden-full-source-line

@Route("login-overlay-basic")
public class LoginOverlayBasic extends Div {

    public LoginOverlayBasic() {
        // tag::snippet[]
        LoginOverlay loginOverlay = new LoginOverlay();
        add(loginOverlay);

        Button login = new Button("Log in");
        login.addClickListener(event -> loginOverlay.setOpened(true));
        // end::snippet[]
        login.addThemeVariants(ButtonVariant.LUMO_PRIMARY);
        add(login);
    }
    public static class Exporter extends DemoExporter<LoginOverlayBasic> {} // hidden-full-source-line
}

File: frontend/demo/component/login/login-overlay-basic.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/button';
import '@vaadin/login';
import { html, LitElement } from 'lit';
import { customElement, state } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/demo/theme';

@customElement('login-overlay-basic')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    applyTheme(root);
    return root;
  }

  @state()
  private loginOpened = false;

  protected override render() {
    return html`
      <!-- tag::snippet[] -->
      <vaadin-button
        theme="primary"
        @click="${() => {
          this.loginOpened = true;
        }}"
      >
        Log in
      </vaadin-button>
      <vaadin-login-overlay
        .opened="${this.loginOpened}"
        @login="${() => {
          this.loginOpened = false;
        }}"
      ></vaadin-login-overlay>
      <!-- end::snippet[] -->
    `;
  }
}
